release-team/hack/sync-bug-triage-github-project-beta.sh

```diff
@@ -0,0 +1,225 @@
+
+
+ORGANIZATION="kubernetes"
+REPOSITORY="kubernetes"
+PROJECT_NUMBER=${GITHUB_PROJECT_BETA_NUMBER}
+MILESTONE=${RELEASE_MILESTONE}
+
+milestone_issue_ids=()
+milestone_pr_ids=()
+
+# function definitions
+function get_field_ids_from_github_beta_project() {
+    if [ $1 == "project-id" ]
+    then
+        query='.data.organization.projectV2.id'
+    elif [ $1 == "type-id" ]
+    then
+        query='.data.organization.projectV2.fields.nodes[] | select(.name== "Type") | .id'
+    elif [ $1 == "issue-id" ]
+    then
+        query='.data.organization.projectV2.fields.nodes[] | select(.name== "Type") | .options[] | select(.name== "Issue") | .id'
+    # for "pr-id"    
+    else
+        query='.data.organization.projectV2.fields.nodes[] | select(.name== "Type") | .options[] | select(.name== "PR") | .id'
+    fi
+    
+    ID="$( gh api graphql -f query='
+                    query($org: String!, $number: Int!) {
+                    organization(login: $org){
+                        projectV2(number: $number) {
+                        id
+                        fields(first:100) {
+                            nodes {
+                            ... on ProjectV2Field {
+                                id
+                                name
+                            }
+                            ... on ProjectV2SingleSelectField {
+                                id
+                                name
+                                options {
+                                    id
+                                    name
+                                }
+                            }
+                            }
+                        }
+                        }
+                    }
+                    }' -f org=${ORGANIZATION} -F number=${PROJECT_NUMBER} --jq "$query")"
+    echo $ID
+}
+
+
+function add_items_to_github_beta_project {
+    	    gh api graphql -f query='
+                    mutation (
+                        $project: ID!
+                        $item: ID!
+                        $type_field: ID!
+                        $option_id: String!
+                    ) {
+                        set_issue_type: updateProjectV2ItemFieldValue(input: {
+                            projectId: $project
+                            itemId: $item
+                            fieldId: $type_field
+                            value: {
+                                singleSelectOptionId: $option_id
+                            }
+                        }) {
+                            projectV2Item {
+                                id
+                            }
+                        }
+                    }' -f project="${PROJECT_ID}" -f item="${1}"  -f type_field="${TYPE_FIELD_ID}" -f option_id="${2}" --silent
+}
+
+echo 'Startig sync...'
+echo -e "[INFO] Fetching the list of open issues and PRs from k/k under the current release milestone, ${MILESTONE}"
+
+## Fetch issues
```

Comment:
This query won't be paginated.
You'll want to specify the `--paginate` flag and have the query take in an `$endCursor` arg.

ex:
``` json
$ gh api graphql --paginate -f query='
  query($endCursor: String) {
    viewer {
      repositories(first: 100, after: $endCursor) {
        nodes { nameWithOwner }
        pageInfo {
          hasNextPage
          endCursor
        }
      }
    }
  }
'
```
from
https://cli.github.com/manual/gh_api

Reply:
Added.